src/client/services/Network.ts

```diff
@@ -0,0 +1,77 @@
+import { Client, Room } from 'colyseus.js'
+import Phaser from 'phaser'
+import { IOfficeState, IPlayer } from '../../types/IOfficeState'
+import { Message } from '../../types/Messages'
+
+export default class Network {
+  private client: Client
+  private room?: Room<IOfficeState>
+  private events: Phaser.Events.EventEmitter
+
+  private _mySessionId!: string
+  set mySessionId(id: string) {
+    this._mySessionId = id
+  }
+  get mySessionId() {
+    return this._mySessionId
+  }
+
+  constructor() {
+    const protocol = window.location.protocol.replace('http', 'ws')
+    const endpoint =
+      process.env.NODE_ENV === 'production'
+        ? `wss://sky-office.herokuapp.com`
+        : `${protocol}//${window.location.hostname}:2567`
+    this.client = new Client(endpoint)
+    this.events = new Phaser.Events.EventEmitter()
+  }
+
+  async join() {
+    this.room = await this.client.joinOrCreate('skyoffice')
+    this.mySessionId = this.room.sessionId
+
+    // new instance added to the players MapSchema
+    this.room.state.players.onAdd = (player: IPlayer, key: string) => {
+      if (key === this.mySessionId) return
+
+      this.events.emit('player-joined', player, key)
```

Comment:
use a enum for event names, to prevent typo

Reply:
Done!